### PR TITLE
Add METL 1D embeddings (Option A)

### PR DESCRIPTION
**Pull request by jahnzh**
## Summary
- Adds METL 1D protein embedding support via optional `metl-pretrained` backend (`pip install pepe-cli[metl]`).
- Model identifiers like `metl-g-20m-1d` route to `METLEmbedder` with `METLDataset`; `[metl]` and `[esm]` extras in packaging.
- Clear errors for missing METL package (`METLPackageRequiredError`) and 3D model ids (`METL3DNotSupportedError`).
- **Limitations:** 1D embeddings only (`per_token`, `mean_pooled`, `substring_pooled`); no logits or attention; 3D METL models rejected.
- Dispatch and integration tests (5 unit + optional `METL_TEST=1` in CI).

## Test plan
- [x] `python -m pytest src/tests/test_metl_dispatch.py`
- [x] `METL_TEST=1 python -m pytest src/tests/test_metl_integration.py`
- [ ] CI on this PR (test workflow including METL integration when enabled)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -35,6 +35,8 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
+metl = ["metl-pretrained @ git+https://github.com/gitter-lab/metl-pretrained.git"]
+esm = ["fair-esm"]
```
[the remark this answers is not in the document]
Comment (jahnzh) [reply pyproject.toml:39]: @copilot+gpt-5.3-codex pyproject.toml doesn't support VCS dependencies I think?
